cli/cmd/vulnerability.go

```diff
@@ -364,13 +394,51 @@ func pollScanStatus(requestID string) error {
 
 		cli.StopProgress()
 		cli.OutputHuman(buildVulnerabilityReportTable(assessment))
+
 		if vulCmdState.Html {
 			return generateVulnAssessmentHTML(assessment)
 		}
+
+		if vulFailureFlagsEnabled() {
+			vulnAssess := vulAssessment{
+				severityRating:        assessment.HighestSeverity(),
+				fixableSeverityRating: assessment.HighestFixableSeverity(),
+				fixableVulCount:       assessment.FixableVulnerabilities,
+			}
+			vulnAssess.validate()
+		}
+
 		return nil
 	}
 }
 
+type vulAssessment struct {
+	severityRating        string
+	fixableSeverityRating string
+	fixableVulCount       int32
+}
+
+func (assessment *vulAssessment) validate() {
+	severityRating, _ := eventSeverityToProperTypes(assessment.severityRating)
+	fixableSeverityRating, _ := eventSeverityToProperTypes(assessment.fixableSeverityRating)
+	threshold, _ := eventSeverityToProperTypes(vulCmdState.FailOnSeverity)
+
+	if vulCmdState.FailOnSeverity == "" && vulCmdState.FailOnFixable && assessment.fixableVulCount > 0 {
+		cli.OutputHuman("\nFixable vulnerabilities found. Exiting with error code %d\n", 9)
+		os.Exit(9)
+	}
+
+	if vulCmdState.FailOnFixable && severityRating <= fixableSeverityRating {
+		cli.OutputHuman("\nFixable vulnerabilities found with threshold \"%s\". Exiting with error code %d\n", vulCmdState.FailOnSeverity, 9)
+		os.Exit(9)
+	}
+
+	if !vulCmdState.FailOnFixable && severityRating <= threshold {
+		cli.OutputHuman("\nVulnerabilities found with threshold \"%s\". Exiting with error code %d\n", vulCmdState.FailOnSeverity, 9)
+		os.Exit(9)
```

Comment:
I am not 100% sure that we want to exit spontaneously in different places of the CLI codebase. I fear we are going to lose track of the places/functions that could exit a command executing and we won't provide end-to-end features and functionalities. For example, the daily version check or telemetry could be lost here. 🤔

Let me think for a min how can we work something out so that we avoid doing these kinds of exit.

Reply:
Additionally, there is already a function you could use to exit with an exit code `exitwithCode()` https://github.com/lacework/go-sdk/blob/main/cli/cmd/root.go#L277

Reply:
I'll change these to exiwithCode in the mean time